app/models/reaction.rb

```diff
@@ -66,7 +66,14 @@ def cached_any_reactions_for?(reactable, user, category)
   # - receiver is the same user as the one who reacted
   # - receive_notification is disabled
   def skip_notification_for?(_receiver)
-    points.negative? || (user_id == reactable.user_id)
+    reactor_id = case reactable
+                 when User
+                   reactable.id
+                 else
+                   reactable.user_id
+                 end
+
+    points.negative? || (user_id == reactor_id)
```

Comment:
I know you just moved this code but since we're touching it already, maybe we can get rid of the unnecessary parenthesis? `==` has a higher precedence than `||` ([Ruby documentation](https://ruby-doc.org/core-2.7.2/doc/syntax/precedence_rdoc.html)).